pdf/pdfsearchmodel.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2015 Caliste Damien.
+ * Contact: Damien Caliste <dcaliste@free.fr>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; version 2 only.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#include "pdfsearchmodel.h"
+
+#include <QRectF>
+
+PDFSearchModel::PDFSearchModel(const QList< QPair<int, QRectF> > &matches, QObject* parent)
+  : QAbstractListModel(parent), m_matches(matches)
+{
+}
+
+PDFSearchModel::~PDFSearchModel()
+{
+}
+
+QHash< int, QByteArray > PDFSearchModel::roleNames() const
+{
+    QHash< int, QByteArray > names;
+    names[Page] = "page";
+    names[Rect] = "rect";
+    return names;
+}
```

Comment:
normally this is done in the class constructor, then here we just just return the roles, otherwise this is constructed every time this function is called, see for example: https://github.com/sailfishos/sailfish-office/blob/master/models/documentlistmodel.cpp#L49 or more organized example: https://github.com/sailfishos/sailfish-office/blob/master/models/documentlistmodel.cpp#L49